modules/sdk-core/src/account-lib/mpc/tss/ecdsa/ecdsa.ts

```diff
@@ -346,21 +346,23 @@ export default class Ecdsa {
    * @returns {Signature}
    */
   constructSignature(shares: SShare[]): Signature {
-    // Every r must match.
-    const r = shares[0]['r'];
-    const isRMatching = shares.map((share) => share['r'] === r).reduce((a, b) => a && b);
+    // Every R must match.
+    const R = shares[0]['R'];
+    const isRMatching = shares.map((share) => share['R'] === R).reduce((a, b) => a && b);
     if (!isRMatching) {
-      throw new Error('r value should be consistent across all shares');
+      throw new Error('R value should be consistent across all shares');
     }
 
     let s = shares.map((share) => hexToBigInt(share['s'])).reduce(Ecdsa.curve.scalarAdd);
+    const recid = (R.slice(0, 2) === '03' ? 1 : 0) ^ (s > Ecdsa.curve.order() / BigInt(2) ? 1 : 0);
```

Comment:
Can we add a small unit test for this new addition `recid` value may be by using the SShares in fixtures and testing whether we get expected `recid` value back

Reply:
@olibeyene @anshuldoshi  you could make use of this for calculating the 'v' value

Reply:
I cant find any unit tests for the mpc library itself which seems very strange. I could just be missing it. Instead I modified the secp256k1 curve implementation no use public key recovery to to verify the signature, so any existing signature verification will automatically test the recid field.

Reply:
https://github.com/BitGo/BitGoJS/tree/master/modules/account-lib/test/unit/mpc/tss/ecdsa tests are in a different module, I understand the confusion